lxd/instance_exec.go

```diff
@@ -519,8 +519,15 @@ func instanceExecPost(d *Daemon, r *http.Request) response.Response {
 	_, ok := post.Environment["PATH"]
 	if !ok {
 		post.Environment["PATH"] = "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
-		if inst.FileExists("/snap") == nil {
-			post.Environment["PATH"] = fmt.Sprintf("%s:/snap/bin", post.Environment["PATH"])
+
+		// Add some additional paths. This directly looks through /proc
+		// rather than use FileExists as none of those paths are expected to be
+		// symlinks and this is much faster than forking a sub-process and
```

Comment:
/run/current-system is a symlink into /nix/store, not sure if that'll work

Reply:
checking whether /run/current-system exists (but checking so it ignores the fact the symlink points to an invalid location) should fix that

could be key->value 

thingToCheck = path

/snap = /snap/bin
/run/current-system = /run/current-system/sw/bin

Reply:
We certainly can make a map of some kind for those checks, though if we can point directly to the absolute target of the symlink, that's likely a better option which will also save a tiny bit of effort to the shell later on.

Reply:
Actually /run/current-system doesn't link to /nix/store but to /nix/store/unique-hash and that unique hash changes everytime, so the best we can do is to check /run/current-system, as everything else is variable

Another idea is to just check if /etc/NIXOS (with exactly that case) exists which is the marker nixOS uses internally to identify if this system is a NixOS installation

(Checking for /nix/store is also possible but nix can be installed independently of nixOS so we'd detect /nix/store on say ubuntu with nix installed aswell)

Reply:
Ok, changed to a map which maps the path to check to the path to add.